engine/src/main/java/org/terasology/world/block/tiles/WorldAtlasImpl.java

```diff
@@ -310,13 +355,14 @@ private BufferedImage generateAtlas(int mipMapLevel, List<BlockTile> tileImages,
         Graphics g = result.getGraphics();
 
         g.setColor(clearColor);
-        g.fillRect(0, 0, size, size);
         for (int index = 0; index < tileImages.size(); ++index) {
             int posX = (index) % tilesPerDim;
             int posY = (index) / tilesPerDim;
             BlockTile tile = tileImages.get(index);
             if (tile != null) {
                 g.drawImage(tile.getImage().getScaledInstance(textureSize, textureSize, Image.SCALE_SMOOTH), posX * textureSize, posY * textureSize, null);
+            } else {
+                g.fillRect(posX * textureSize, posY * textureSize, textureSize, textureSize);
```

Comment:
Fill each tile separately so that transparency on the original tile does not get blended with a base color.